src/objectid.ts

```diff
@@ -7,10 +7,12 @@ import { NumberUtils } from './utils/number_utils';
 // Unique sequence for the current process (initialized on first use)
 let PROCESS_UNIQUE: Uint8Array | null = null;
 
+/** ObjectId hexString cache @internal */
+const __idCache = new WeakMap(); // TODO convert this to #__id private field when target updated to ES2022
+
 /** @public */
 export interface ObjectIdLike {
   id: string | Uint8Array;
-  __id?: string;
   toHexString(): string;
```

Comment:
We do need to keep this, even though it won't be present on new object ids, because the interface itself is also public.  We can remove this in v7